### ISSUE TITLE
SPLASH: mockups for expanded search

### ISSUE THREAD

**Issue by mapsam** [codeforamerica/rva-screening-ui-prototypes#14]
Entry point for _assisters_ and _screeners_ is to "add a patient" OR "search for a patient" so let's try and combine these into the first landing page, allowing you to do both at once.

**Comment by mapsam** [comment]
Expanded search mockups are underway!

<img width="1043" alt="screen shot 2015-07-13 at 2 49 14 pm" src="https://cloud.githubusercontent.com/assets/1943001/8661718/63f472a2-296e-11e5-87e0-06c592cd241c.png">

**Comment by mapsam** [comment]
What do you think about adding a search button @bengolder? You mentioned this search automatically as the user enters information. Do you think that happens on every `keyUp` event or something similar?

**Comment by mapsam** [comment]
Example showing search results and an "add new patient" button at the bottom of the list. **Where** the add new patient button goes is still up for discussion - either the bottom or the top of the list!

<img width="903" alt="screen shot 2015-07-13 at 3 45 32 pm" src="https://cloud.githubusercontent.com/assets/1943001/8662632/a4f78368-2976-11e5-86bd-3eaf8863ea58.png">

**Comment by bengolder** [comment]
Yeah, I think we should respond to keyUp events. Though we will want some stronger visual feedback rather than just immediately adding divs. Something should probably flash, fade, or move to indicate that a search is happening. A button seems fine and redundant.

**Comment by mapsam** [comment]
Version of search with no results:

<img width="881" alt="screen shot 2015-07-13 at 3 56 30 pm" src="https://cloud.githubusercontent.com/assets/1943001/8662740/cb7b65ee-2977-11e5-9941-cba937e95d86.png">

**Comment by mapsam** [comment]
re: visual feedback on the search

![rva-screener-searching](https://cloud.githubusercontent.com/assets/1943001/8662806/6d86f8bc-2978-11e5-974a-3db71fb0535e.gif)

**Comment by mapsam** [comment]
Does the expanded search mean we can get rid of the "find patient in your network" and "create patient" navigation buttons?

**Comment by mapsam** [comment]
An alternative to expanded search with the "create new patient" button the right, and results on the left. Notice the results show _In Network_ and _Out of Network_ patients.

<img width="1159" alt="screen shot 2015-07-16 at 9 19 35 am" src="https://cloud.githubusercontent.com/assets/1943001/8728764/ead28eec-2b9b-11e5-8d95-80374f1a6bd0.png">

**Comment by bengolder** [comment]
A few thoughts:
- I like the differentiation of results that are "in-network" vs. "outside of network", but I think the language is unclear. Perhaps this subtitle should be reworded as "RCHD Resource Center Patients" and "Patients at other services".
- Yes I think expanded search means we can get rid of "find patient in your network" and "create patient" 
- I'm unsure about the relative prominence between search results and creating a patient. I think many of our users will be entering new patients at least 80% of the time, especially in the first year of the app. I think "Create a new Patient Record" should be a prominent, simple action, perhaps of greater visual rank than the search results.
- If side by side, how should it lay out on mobile?
- I think it might be nice to have all these results more compact, including the create patient option. I'm imagining that a user should be able to tab through the results, maybe as a dropdown.

**Comment by mapsam** [comment]
Thanks for the clear feedback! 

**Wording**: I think it's definitely best to make sure that when we are talking about "patients" the language states _which_ patients, just to make sure the information is 100% clear.

**Menu items**: :+1: 

**Results vs. Create Patient**: Creating will definitely happen a lot in the beginning, so we'll want to make sure it is the first thing that pops up.

**Side by side**: They can stack on mobile - but in regards to the above prioritization of search, we could push it to the left-hand side instead of the right so it stacks on top instead of beneath the results. (FWIW I'm not totally sold on the side-by-side, just did a mockup of it since it was quick!)

**Compactness**: definitely possible. Would be good to build in options for lists such as `.list_compact` vs `.list_comfortable` to be used throughout the app.

**Comment by mapsam** [comment]
Working with [Defiant.js](http://www.defiantjs.com/) to filter through a given JSON and search for results using `XPath` queries.

![rva-screener-defiantsearch](https://cloud.githubusercontent.com/assets/1943001/8891049/f4333828-32cc-11e5-8761-65768d397f74.gif)

This uses a rendering template as well, which makes for passing the JSON results into HTML quite simple. Here's the patient list template:

``` HTML
<script type="defiant/xsl-template">
    <xsl:template name="search_template">
        <xsl:for-each select="//patients">
            <li class="list_row">
              <a href="#">
                <span class="list_row_item list_row_name"><xsl:value-of select="name"/></span>
                <span class="list_row_item list_row_dob"><xsl:value-of select="dob"/></span>
                <span class="list_row_item list_row_edits">Richmond Resource Center</span>
              </a>
            </li>
        </xsl:for-each>
    </xsl:template>
</script>
```

**Comment by mapsam** [comment]
Rendering different templates based on search results:

![rva-screener-defiantsearch2](https://cloud.githubusercontent.com/assets/1943001/8891194/68bc6708-32d4-11e5-8a00-7fe7e21f45f2.gif)
